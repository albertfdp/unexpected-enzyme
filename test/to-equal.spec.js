import expect from './unexpected-enzyme';
import React from 'react';
import PropTypes from 'prop-types';

class Fixture extends React.Component {
  render() {
    return (
      <div>
        <div id="parent" className="old friendly">
          <div id="child">{this.props.children}</div>
        </div>
      </div>
    );
  }
}

Fixture.propTypes = {
  children: PropTypes.node
};

describe('to-equal', () => {
  describe('comparing react elements', () => {
    it('passes when the given react elements are equals', () =>
      expect(
        <div>
          <div id="parent" className="old friendly">
            <div id="child">Children</div>
          </div>
        </div>,
        'to equal',
        <div>
          <div id="parent" className="old friendly">
            <div id="child">Children</div>
          </div>
        </div>
      ));

    it('fails when the actual does not match the expected', () =>
      expect(
        () =>
          expect(
            <div>
              <div id="parent" className="old friendly">
                <div id="child">Children</div>
              </div>
            </div>,
            'to equal',
            <div>
              <div id="parents" className="old friendly">
                <div id="child">Kids</div>
              </div>
            </div>
          ),
        'with error matching snapshot'
      ));

    describe('when negated', () => {
      it('passes when actual is not equal the expected', () =>
        expect(
          <div>
            <div id="parent" className="old friendly">
              <div id="child">Children</div>
            </div>
          </div>,
          'not to equal',
          <div>
            <div id="parents" className="old friendly">
              <div id="childs">Kids</div>
            </div>
          </div>
        ));

      it('fails when actual is equal the expected', () =>
        expect(
          () =>
            expect(
              <div>
                <div id="parent" className="old friendly">
                  <div id="child">Children</div>
                </div>
              </div>,
              'not to equal',
              <div>
                <div id="parent" className="old friendly">
                  <div id="child">Children</div>
                </div>
              </div>
            ),
          'with error matching snapshot'
        ));
    });
  });
});
