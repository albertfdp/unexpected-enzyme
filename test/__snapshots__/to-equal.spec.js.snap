// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`to-equal comparing react elements fails when the actual does not match the expected 1`] = `
"expected
<div>
  <div id=\\"parent\\" className=\\"old friendly\\"><div id=\\"child\\">Children</div></div>
</div>
to equal
<div>
  <div id=\\"parents\\" className=\\"old friendly\\"><div id=\\"child\\">Kids</div></div>
</div>

<div>
  <div id=\\"parent\\" // expected 'parent' to equal 'parents'
                   //
                   // -parent
                   // +parents
     className=\\"old friendly\\">
    <div id=\\"child\\">
      Children // -Children
               // +Kids
    </div>
  </div>
</div>"
`;

exports[`to-equal comparing react elements when negated fails when actual is equal the expected 1`] = `
"expected
<div>
  <div id=\\"parent\\" className=\\"old friendly\\"><div id=\\"child\\">Children</div></div>
</div>
not to equal
<div>
  <div id=\\"parent\\" className=\\"old friendly\\"><div id=\\"child\\">Children</div></div>
</div>"
`;
